static/js/xmleditor/xsd2json.js

```diff
@@ -466,25 +480,77 @@ SchemaManager.prototype.resolveSchema = function(schema, name) {
 	return this;
 };
 
-SchemaManager.prototype.mergeType = function(base, type) {
+SchemaManager.prototype.mergeType = function(base, type, mergeMode) {
 	for (var key in type) {
 		if (type.hasOwnProperty(key)) {
 			var value = type[key];
 			if (value != null && base[key] == null){
 				base[key] = value;
-			} else if ($.isArray(value) && $.isArray(type[key])){
-				base[key] = base[key].concat(value);
+			} else if ($.isArray(value) && $.isArray(base[key])){
+				base[key] = this.mergeTypeArray(base[key], value, key, mergeMode);
 			}
 		}
 	}
 };
 
+SchemaManager.prototype.mergeTypeArray = function(baseArray, typeArray, key, mergeMode) {
+	var mergeFunction = this['mergeType_' + key + '_' + mergeMode];
+	if (mergeFunction === undefined) {
+		throw Error('Invalid definition key ' + JSON.stringify(key) + ' or merge mode ' + JSON.stringify(mergeMode));
+	}
+	return mergeFunction.call(this, baseArray, typeArray);
+};
+
```

Comment:
this method is gluing together components to form a function name and then calling it, with a bit of validation so that if something goes awry it won't be as hard to debug.  I think vog's intention was to not have a mountain of if statements to determine which merge function to call, to instead have this thing work it out.  I guess it's a very javascripty approach to the problem